### PR TITLE
add strikethrough font effect support

### PR DESCRIPTION
**Pull request by faywong**

### CODE REVIEW

File: src/tb/tb_style_edit.cpp [hunk 1]
```diff
@@ -1262,7 +1264,11 @@ void TBTextFragment::Paint(int32_t translate_x, int32_t translate_y, TBTextProps
 		int line_h = font->GetHeight() / 16;
 		line_h = MAX(line_h, 1);
 		listener->DrawRectFill(TBRect(x, y + GetBaseline(font) + 1, GetWidth(font), line_h), color);
-	}
```
Comment (tesch1) [inline src/tb/tb_style_edit.cpp:1265]: Could text be both underlined and striken?

**Review by tesch1 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Could you please also add a change to the demo program to show the new functionality?

File: src/tb/tb_style_edit.cpp (hunk 1 above)
Comment (faywong) [reply src/tb/tb_style_edit.cpp:1265]: There is no use case of combination of underlined/strikethrough/upperline.

**Comment by faywong** [comment]
> Thank you for your contribution!
> 
> Could you please also add a change to the demo program to show the new functionality?

It lies in the line 31 of Demo/demo01/ui_resources/test_textwindow.tb.txt

`` <s>Strikethrough</s> ``

File: src/tb/tb_style_edit.cpp (hunk 1 above)
Comment (tesch1) [reply src/tb/tb_style_edit.cpp:1265]: Not sure how to verify that?

iirc, it works in html?  What should a user expect to happen w/ nested tags?

https://www.w3schools.com/html/tryit.asp?filename=tryhtml_intro

Comment (faywong) [reply src/tb/tb_style_edit.cpp:1265]: In css, it's expressed in css rule: [text-decoration](https://www.w3schools.com/cssref/tryit.php?filename=trycss_text-decoration). 

There's ``BBCode`` markup of  [Strikethrough](https://www.bbcode.org/making-strikethrough-text-with-bbcode.php).  [Godot Engine](https://docs.godotengine.org/en/stable/tutorials/ui/bbcode_in_richtextlabel.html) adopted the BBCode way.

 In my opinion, it is more like a convention, in actual world, there's seldom a need to combine these text-decorations. 





<br class="Apple-interchange-newline">

**Pull request closed without merging**